src/main/java/com/tickets/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.tickets.dao.impl;
+
+import com.tickets.dao.CinemaHallDao;
+import com.tickets.exception.DataProcessingException;
+import com.tickets.lib.Dao;
+import com.tickets.model.CinemaHall;
+import com.tickets.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Cinema Hall entity: " + cinemaHall, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllCinemaHalls = session
+                    .createQuery("from cinema_hall ", CinemaHall.class);
```

Comment:
I think it is better to use class name and not the table name here - so that we could understand that in HQL we can refer to classes and their fields